iOS_SDK/OneSignalSDK/OneSignalInAppMessages/Controller/OSMessagingController.m

```diff
@@ -248,16 +253,55 @@ - (void)getInAppMessagesFromServer:(NSString *)subscriptionId {
         [self updateInAppMessagesFromCache];
         return;
     }
-    
-    OSRequestGetInAppMessages *request = [OSRequestGetInAppMessages withSubscriptionId:subscriptionId];
-    [OneSignalCoreImpl.sharedClient executeRequest:request onSuccess:^(NSDictionary *result) {
+
+    OSConsistencyManager *consistencyManager = [OSConsistencyManager shared];
+    NSString *onesignalId = OneSignalUserManagerImpl.sharedInstance.onesignalId;
+
+    if (!onesignalId) {
+        [OneSignalLog onesignalLog:ONE_S_LL_VERBOSE message:@"Failed to get in app messages due to no OneSignal ID"];
+        return;
+    }
+
+    OSIamFetchReadyCondition *condition = [OSIamFetchReadyCondition sharedInstanceWithId:onesignalId];
+    NSString *rywToken = [consistencyManager registerCondition:condition forId:onesignalId];
+
+    NSNumber *sessionDuration = @([OSSessionManager getTimeFocusedElapsed]);
+
+    // Initial request
+    [self attemptFetchWithRetries:subscriptionId
+                           rywToken:rywToken
+                  sessionDuration:sessionDuration
+                        attempts:@0 // Starting with 0 attempts
+                     retryLimit:nil]; // Retry limit to be set dynamically on first failure
+}
+
+- (void)attemptFetchWithRetries:(NSString *)subscriptionId
+                         rywToken:(NSString *)rywToken
+                sessionDuration:(NSNumber *)sessionDuration
+                       attempts:(NSNumber *)attempts
+                     retryLimit:(NSNumber *)retryLimit {
+
+    if ([sessionDuration doubleValue] == -1) {
+        sessionDuration = @0;
+    }
+
+    // Create the request with the current attempt count
+    OSRequestGetInAppMessages *request = [OSRequestGetInAppMessages withSubscriptionId:subscriptionId
+                                                                      withSessionDuration:sessionDuration
+                                                                      withRetryCount:attempts
+                                                                          withRywToken:rywToken];
```

Comment:
Check the aligning of the parameters, this is in a few different places in the PR. The way I get them to align is backspace to the left and then tab over and it auto aligns.